org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/scan/DeviceRequest.java

```diff
@@ -11,55 +11,57 @@
  *******************************************************************************/
 package org.eclipse.scanning.api.event.scan;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.LinkedHashSet;
+import java.util.Map;
 
 import org.eclipse.scanning.api.ModelValidationException;
 import org.eclipse.scanning.api.ValidationException;
 import org.eclipse.scanning.api.annotation.ui.DeviceType;
+import org.eclipse.scanning.api.device.IAttributableDevice;
 import org.eclipse.scanning.api.event.EventException;
 import org.eclipse.scanning.api.event.IdBean;
+import org.eclipse.scanning.api.malcolm.attributes.IDeviceAttribute;
 import org.eclipse.scanning.api.points.IPosition;
 
 /**
- * 
+ *
  * Object used to query which devices are available from a given broker.
  * The solstice server for instance asks the IRunnableDeviceService what is available
  * and returns a list of devices and their configuration.
- * 
+ *
  * Servlet may also be used to configure a given device and return the result, set its value
  * get its value. It can be used for Scannables and Detectors.
- * 
+ *
  * <pre>
  * Usage:
- * 1. Set nothing, post returns list of DeviceInformation for all devices. 
+ * 1. Set nothing, post returns list of DeviceInformation for all devices.
  * 2. Set the device name, post returns the device with this name.               IRunnableDeviceService.getRunnableDevice()
- * 3. Set name and model, named device is retrieved and configured.        
+ * 3. Set name and model, named device is retrieved and configured.
  * 4. Set the device model and the configure boolean, get a new device created.  IRunnableDeviceService.createRunnableDevice()
  * 5. Set the device action and the device name to call specific methods.
- * 
+ *
  * </pre>
- * 
+ *
  * TODO The data in this class has become a little overloaded. We could have one class for scannables

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)